python/src/aiconfig/editor/client/src/components/SettingsPropertyRenderer.tsx

```diff
@@ -329,6 +346,68 @@ export default function SettingsPropertyRenderer({
     }
     case "object": {
       const requiredFields = new Set<string>(property.required ?? []);
+      // useEffect(() => {
+      //   requiredFields.forEach((fieldName) => {
+      //     onUpdateMissingRequiredFields(fieldName, propertyValue);
+      //   });
+      // }, [onUpdateMissingRequiredFields, propertyValue, requiredFields]);
+      /* 
+Warning: Cannot update a component (`PromptContainer`) while rendering a different component (`SettingsPropertyRenderer`). To locate the bad setState() call inside `SettingsPropertyRenderer`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
```

Comment:
See L76 for more context on what I initially tried to do here and why I had to move it there instead